lib/experimental/class-gutenberg-rest-comment-controller.php

```diff
@@ -110,6 +110,36 @@ public function create_item_permissions_check( $request ) {
 			);
 		}
 
+		if ( ! $this->check_post_type_supports_block_comments( $post->post_type ) ) {
+			return new WP_Error(
+				'rest_comment_block_comments_not_supported',
+				__( 'Sorry, this post type does not support block comments.', 'gutenberg' ),
+				array( 'status' => 403 )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check if post type supports block comments.
+	 *
+	 * @param string $post_type Post type name.
+	 * @return bool True if post type supports block comments, false otherwise.
+	 */
+	private function check_post_type_supports_block_comments( $post_type ) {
+		$supports = get_all_post_type_supports( $post_type );
+		if ( ! isset( $supports['editor'] ) ) {
+			return false;
+		}
+		if ( ! is_array( $supports['editor'] ) ) {
+			return false;
+		}
+		foreach ( $supports['editor'] as $item ) {
+			if ( is_array( $item ) && isset( $item['block-comments'] ) && true === $item['block-comments'] ) {
+				return true;
+			}
+		}
```

Comment:
I have never used or tested nested support keys, but I didn't realize it would be so complicated to check for support.

Reply:
You can also use something like this: `isset` supports multi-dimensional arrays.

```php
isset( $supports['editor']['block-comments'] ) && true === $supports['editor']['block-comments']
```

P.S. Now that we've started using more nested support keys, maybe it's time to update the `post_type_supports` function to make checks easier.

Reply:
Oh, you're right. That's an annoying data format.